Error loading fonts from directories with spaces in Diffenator2

When attempting to load fonts from directories with spaces in their names, Diffenator2 generates a "ninja: build stopped: subcommand failed." error.

Steps to reproduce:
1. Create a directory with a space in its name (e.g. "My Fonts").
2. Place font files in the directory.
3. Open Diffenator2 and attempt to load the fonts from the directory with spaces.

Expected result:
Fonts should load successfully from directories with spaces in their names.

Actual result:
Diffenator2 generates a "ninja: build stopped: subcommand failed." error when attempting to load fonts from directories with spaces in their names.

Workaround:
Users can rename the directory to remove the spaces or move the font files to a directory without spaces to load the fonts into Diffenator2 successfully.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -32,3 +32,5 @@ jobs:
       - name: Run Tests
         run: |
           pytest tests/
+        env:
+          GH_TOKEN: ${{ github.token }}

--- a/src/diffenator2/__init__.py
+++ b/src/diffenator2/__init__.py
@@ -66,7 +66,7 @@ def _ninja_proof(
     # Setup build
     w.comment("Build rules")
     variables = dict(
-        fonts=[os.path.abspath(f.ttFont.reader.file.name) for f in fonts],
+        fonts=[f'"{os.path.abspath(f.ttFont.reader.file.name)}"' for f in fonts],
         styles=styles,
         out=out_s,
         pt_size=pt_size
@@ -182,8 +182,8 @@ def _ninja_diff(
     if diffbrowsers:
         diffbrowsers_out = os.path.join(out, "diffbrowsers")
         db_variables = dict(
-            fonts_before=[os.path.abspath(f.ttFont.reader.file.name) for f in fonts_before],
-            fonts_after=[os.path.abspath(f.ttFont.reader.file.name) for f in fonts_after],
+            fonts_before=[f'"{os.path.abspath(f.ttFont.reader.file.name)}"' for f in fonts_before],
+            fonts_after=[f'"{os.path.abspath(f.ttFont.reader.file.name)}"' for f in fonts_after],
             styles=styles,
             out=diffbrowsers_out,
             pt_size=pt_size
@@ -198,8 +198,8 @@ def _ninja_diff(
             coords = new_style.coords
             style = new_style.name.replace(" ", "-")
             diff_variables: dict[str,str] = dict(
-                font_before=old_style.font.ttFont.reader.file.name,
-                font_after=new_style.font.ttFont.reader.file.name,
+                font_before=f'"{old_style.font.ttFont.reader.file.name}"',
+                font_after=f'"{new_style.font.ttFont.reader.file.name}"',
                 out=style,
                 threshold=str(threshold),
             )

--- a/tests/data/ninja_files/diff-filter-styles.txt
+++ b/tests/data/ninja_files/diff-filter-styles.txt
@@ -15,26 +15,26 @@ rule diffenator-inst
 # Build rules
 build out/diffbrowsers: diffbrowsers
   fonts_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   fonts_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   styles = instances
   out = out/diffbrowsers
   pt_size = 20
   filters = Medium|ExtraBold
 build out/ExtraBold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = ExtraBold
   threshold = 0.9
   coords = wght=800.0
 build out/Medium: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Medium
   threshold = 0.9
   coords = wght=500.0

--- a/tests/data/ninja_files/diff-imgs.txt
+++ b/tests/data/ninja_files/diff-imgs.txt
@@ -15,57 +15,57 @@ rule diffenator-inst
 # Build rules
 build out/diffbrowsers: diffbrowsers
   fonts_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   fonts_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   styles = instances
   out = out/diffbrowsers
   pt_size = 20
 build out/Black: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Black
   threshold = 0.9
   coords = wght=900.0
 build out/Bold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Bold
   threshold = 0.9
   coords = wght=700.0
 build out/ExtraBold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = ExtraBold
   threshold = 0.9
   coords = wght=800.0
 build out/Medium: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Medium
   threshold = 0.9
   coords = wght=500.0
 build out/Regular: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Regular
   threshold = 0.9
   coords = wght=400.0
 build out/SemiBold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = SemiBold
   threshold = 0.9
   coords = wght=600.0

--- a/tests/data/ninja_files/diff-standard.txt
+++ b/tests/data/ninja_files/diff-standard.txt
@@ -15,57 +15,57 @@ rule diffenator-inst
 # Build rules
 build out/diffbrowsers: diffbrowsers
   fonts_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   fonts_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   styles = instances
   out = out/diffbrowsers
   pt_size = 20
 build out/Black: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Black
   threshold = 0.9
   coords = wght=900.0
 build out/Bold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Bold
   threshold = 0.9
   coords = wght=700.0
 build out/ExtraBold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = ExtraBold
   threshold = 0.9
   coords = wght=800.0
 build out/Medium: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Medium
   threshold = 0.9
   coords = wght=500.0
 build out/Regular: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Regular
   threshold = 0.9
   coords = wght=400.0
 build out/SemiBold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = SemiBold
   threshold = 0.9
   coords = wght=600.0

--- a/tests/data/ninja_files/diff-threshold.txt
+++ b/tests/data/ninja_files/diff-threshold.txt
@@ -15,57 +15,57 @@ rule diffenator-inst
 # Build rules
 build out/diffbrowsers: diffbrowsers
   fonts_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   fonts_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   styles = instances
   out = out/diffbrowsers
   pt_size = 20
 build out/Black: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Black
   threshold = 0.01
   coords = wght=900.0
 build out/Bold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Bold
   threshold = 0.01
   coords = wght=700.0
 build out/ExtraBold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = ExtraBold
   threshold = 0.01
   coords = wght=800.0
 build out/Medium: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Medium
   threshold = 0.01
   coords = wght=500.0
 build out/Regular: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = Regular
   threshold = 0.01
   coords = wght=400.0
 build out/SemiBold: diffenator-inst
   font_before = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   font_after = \$
-      .*/tests/data/MavenPro\[wght\].subset.mod.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.mod.ttf"
   out = SemiBold
   threshold = 0.01
   coords = wght=600.0

--- a/tests/data/ninja_files/proof-filter-styles.txt
+++ b/tests/data/ninja_files/proof-filter-styles.txt
@@ -7,7 +7,7 @@ rule proofing
 # Build rules
 build out: proofing
   fonts = $
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   styles = instances
   out = out/diffbrowsers
   pt_size = 20

--- a/tests/data/ninja_files/proof-imgs.txt
+++ b/tests/data/ninja_files/proof-imgs.txt
@@ -6,7 +6,7 @@ rule proofing
 # Build rules
 build out: proofing
   fonts = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   styles = instances
   out = out/diffbrowsers
   pt_size = 20

--- a/tests/data/ninja_files/proof-standard.txt
+++ b/tests/data/ninja_files/proof-standard.txt
@@ -6,7 +6,7 @@ rule proofing
 # Build rules
 build out: proofing
   fonts = \$
-      .*/tests/data/MavenPro\[wght\].subset.ttf
+      ".*/tests/data/MavenPro\[wght\].subset.ttf"
   styles = instances
   out = out/diffbrowsers
   pt_size = 20

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -29,5 +29,10 @@ def test_download_google_fonts_family_not_existing():
 def test_download_latest_github_release():
     from diffenator2.utils import download_latest_github_release
     with tempfile.TemporaryDirectory() as tmp:
-        files = download_latest_github_release("googlefonts", "Gulzar", tmp)
+        files = download_latest_github_release(
+            "googlefonts",
+            "Gulzar",
+            tmp,
+            github_token=os.environ.get("GH_TOKEN")
+        )
         assert len(files) != 0

